openedx/core/djangoapps/content/course_overviews/tests/test_api.py

```diff
@@ -43,6 +45,22 @@ def test_get_course_overview_or_none_missing(self):
         retrieved_course_overview = get_course_overview_or_none(course_run_key)
         assert retrieved_course_overview is None
 
+    def test_get_course_overview_or_404(self):
```

Comment:
Nit: add `_success` or something to the end of this. Or alternatively, to keep the generic name, just wrap your other test into this one and test both code paths here.